### PR TITLE
added region to gcp cloud

### PR DESCRIPTION
**Pull request by tomersa**

### CODE REVIEW

File: granulate_utils/metadata/cloud.py [hunk 1]
```diff
@@ -103,6 +107,13 @@ def get_gcp_metadata() -> Optional[GcpInstanceMetadata]:
     )
 
 
+def get_gcp_region_from_zone(zone: str) -> str:
+    zone = zone.rsplit('/', 1)[-1]
```
Comment (Jongy) [inline granulate_utils/metadata/cloud.py:111]: What's the operation you are doing here? How does it look exactly? Give a sample comment on input/output when doing string manipulations

File: granulate_utils/metadata/cloud.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ class GcpInstanceMetadata(InstanceMetadataBase):
     instance_id: str
     image_id: str
     name: str
+    region: str = None
```
Comment (Jongy) [inline granulate_utils/metadata/cloud.py:47]: Why `= None`?

File: granulate_utils/metadata/cloud.py (hunk 1 above)
Comment (tomersa) [reply granulate_utils/metadata/cloud.py:111]: An example of zone would be: `projects/234/zones/us-east1-a`
So,
`get_gcp_region_from_zone('projects/234/zones/us-east1-a')` -> returns
`us-east1`

File: granulate_utils/metadata/cloud.py (hunk 2 above)
Comment (tomersa) [reply granulate_utils/metadata/cloud.py:47]: Actually, I can remove it. It's a leftover.

File: granulate_utils/metadata/cloud.py (hunk 1 above)
Comment (Jongy) [reply granulate_utils/metadata/cloud.py:111]: Comment in code pls

**Review by Jongy — CHANGES_REQUESTED** [review]
Just fix lint

**Review by ziv-bar — CHANGES_REQUESTED** [review]
Nice fix, please take a look at the comments below :)
1. This fix commits the change in sAgent. This means we will have to upgrade every `gcp` sAgent now, which I understand is a big deal.
2. The original `zone` parsing happened in `gServer` (probably for the reason above), I suggest we move the logic to `dServer` instead of sAgent.
3. The `gServer` cloud info code does two parse operations to `zone`:
gserver_cloud_metadata_parsing png
* Remove the "project..." prefix from`zone`
* Create the `region` key from the new zone.
I think that if we make a change, we should implement both of the parse operations on `zone`... Same way as the gServer.

**Comment by tomersa** [comment]
> Nice fix, but perhaps the change should be made in a different place.

As discussed given the דיחפות of the situation we will solve it in here.

**Comment by Jongy** [comment]
I think this is the right place to add - this way it's also available to gProfiler

**Comment by tomersa** [comment]
After a discussion with @Tamir-Mizrachi I'm going to move it to dServer. So ignore closing this PR

**Pull request closed without merging**